queue/queue.py

```diff
@@ -11,53 +11,60 @@ def __init__(self, head, tail):
     self.tail = tail
     self.length = 0
 
-    def add_to_tail(self, value):
-      n = Node(value)
+  def add_to_tail(self, value):
+    n = Node(value)
+    if not self.head:
+      self.head = n
+      self.tail = n
+    else:
       self.tail.next_node = n
       self.tail = n 
-      self.length += 1
+    self.length += 1
 
 
-    def add_to_head(self, value):
-      n = Node(value)
+  def add_to_head(self, value):
+    n = Node(value)
+    if not self.head:
+      self.head = n
+      self.tail = n
+    else:
       n.next_value = self.head
       self.head = n
-      self.length += 1
-
-    def remove_head(self):
-      if not self.head:
-        return None
-      if self.length == 1:
-        self.head = None
-        self.tail = None
-        self.length -= 1
-        return self.head
-      else:
-        current_head = self.head
-        self.head = self.head.next_node
-        self.length -= 1
-        return current_head
+    self.length += 1
 
+  def remove_head(self):
+    if not self.head:
+      return None
+    if self.length == 1:
+      self.head = None
+      self.tail = None
+      self.length -= 1
+      return self.head
```

Comment:
this will return None, but line 47 will return a node. Will you always return None, or will you always return the removed Node? either is fine, but it should be the same for both cases.